static/js/public/snap-details/channelMap.js

```diff
@@ -1,294 +1,425 @@
       }
 
-      iframe = document.createElement('iframe');
-      iframe.className = 'js-snap-open-frame';
-      iframe.style.position = 'absolute';
-      iframe.style.top = '-9999px';
-      iframe.style.left = '-9999px';
-      iframe.src = `snap://${name}`;
-      document.body.appendChild(iframe);
-
-      if (typeof ga !== 'undefined') {
-        // The first attempt should be counted towards the 'intent'
-        let label = 'Snap install intent';
-        let value = `${name}`;
-
-        // Subsequent attempts should still be tracked, but not as 'intent'
-        if (attempt > 1) {
-          label = 'Snap install click';
-          value += ` - click ${attempt}`;
-        }
+      ga('gtm1.send', {
+        hitType: 'event',
+        eventCategory: 'Snap details',
+        eventAction: 'Click view in desktop store button',
+        eventLabel: label,
+        eventValue: value
+      });
+    }
 
-        ga('gtm1.send', {
-          hitType: 'event',
-          eventCategory: 'Snap details',
-          eventAction: 'Click view in desktop store button',
-          eventLabel: label,
-          eventValue: value
-        });
-      }
+    this.openDesktopAttempt += 1;
+  }
 
-      attempt += 1;
-    }
-  });
-}
+  selectScreen(screenEl) {
+    const selected = screenEl.getAttribute('aria-selected');
+    const currentlySelected = screenEl.parentNode.querySelector('.is-open');
 
-export default function initChannelMap(el, packageName, channelMapData) {
-  initOpenSnapButtons();
+    if (currentlySelected) {
+      currentlySelected.classList.remove('is-open');
+    }
 
-  const channelMapEl = document.querySelector(el);
-  const channelOverlayEl = document.querySelector('.p-channel-map-overlay');
+    if (!selected) {
+      screenEl.classList.add('is-open');
+      screenEl.setAttribute('aria-selected', "true");
+    }
+  }
 
-  initTabs(channelMapEl);
+  slideToVersions(clickEl) {
+    const slides = clickEl.closest('.p-channel-map__slides');
+    slides.classList.add('show-left');
+    slides.classList.remove('show-right');
+  }
 
-  let closeTimeout;
+  slideToInstructions(clickEl) {
+    // Add content to the right slide area
+    this.writeInstallInstructions(clickEl.dataset.channel);
 
-  // init open/hide buttons
-  const openChannelMap = (event) => {
-    const openButton = event.target.closest('.js-open-channel-map');
+    const slides = clickEl.closest('.p-channel-map__slides');
+    slides.classList.add('show-right');
+    slides.classList.remove('show-left');
+  }
 
-    if (openButton) {
-      // open tab based on button click (or install tab by default)
-      const openTabName = openButton.getAttribute('aria-controls') || 'channel-map-tab-install';
+  writeInstallInstructions(channel) {
+    let paramString = '';
 
-      // clear hiding animation if it's still running
-      clearTimeout(closeTimeout);
+    if (channel.indexOf('latest') === 0) {
+      if (channel.indexOf('stable') !== 7) {
```

Comment:
I think I finally figure out that you are checking here if track is `latest` and channel `stable` in `'latest/stable'` string, but figuring it out from numbers 0 and 7 was not that easy. At least a bit of comment would be nice for our future selves.